configuration/pih/liquibase/sql/create_functions.sql

```diff
@@ -72,7 +72,9 @@ BEGIN
 	WHERE voided = 0
 	  AND concept_id = _conceptID
 	  AND locale = _locale
-	  AND concept_name_type = 'FULLY_SPECIFIED';
+ 	order by locale_preferred desc, ISNULL(concept_name_type) asc, 
+	field(concept_name_type,'FULLY_SPECIFIED','SHORT')
+	limit 1;
 
```

Comment:
@dmdesimone - I think this approach looks fine.  That said, the previous function was written under the assumption that there was always a Fully Specific name in each locale, and this could be relied upon to be there.  If that's not true, we likely need to do more to ensure there is _always_ some sort of name returned for a concept, and that likely requires looking beyond the given locale.  

What I'd say is that we should move locale out of a constraint and into another condition in the order by clause.  So, the first order by criteria would be something like if(_locale = locale, 0, 1).

This way, we basically get the first concept name for a concept, using our preferred order, but don't return null if a concept happens to be missing a name in the locale of choice.

Reply:
This is a good idea.  Extending this, I'd like to put if(locale='en',0,1) as the second sort criteria so it prioritizes English if there is no name in the passed-in locale.